test/e2e/service.go

```diff
@@ -592,3 +594,188 @@ var _ = ginkgo.Describe("Service Hairpin SNAT", func() {
+spec:
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+  type: LoadBalancer
+`)
+		if err := ioutil.WriteFile(loadBalancerYaml, []byte(loadBalancerServiceConfig), 0644); err != nil {
+			framework.Failf("Unable to write CRD config to disk: %v", err)
+		}
+		framework.Logf("Create the Load Balancer configuration")
+		framework.RunKubectlOrDie("default", "create", "-f", loadBalancerYaml)
+	})
+
+	ginkgo.AfterEach(func() {
+		framework.Logf("Delete the Load Balancer configuration")
+		framework.RunKubectlOrDie("default", "delete", "-f", loadBalancerYaml)
+		defer func() {
+			if err := os.Remove(loadBalancerYaml); err != nil {
+				framework.Logf("Unable to remove the CRD config from disk: %v", err)
+			}
+		}()
+	})
+
+	ginkgo.It("Should ensure load balancer service works with pmtu", func() {
+
+		err := framework.WaitForServiceEndpointsNum(f.ClientSet, namespaceName, svcName, 4, time.Second, time.Second*120)
+		framework.ExpectNoError(err, fmt.Sprintf("service: %s never had an endpoint, err: %v", svcName, err))
+
+		numberOfETPRules := pokeIPTableRules(backendNodeName, "OVN-KUBE-EXTERNALIP")
+		framework.ExpectEqual(numberOfETPRules, 4)
+
+		ginkgo.By("by sending a TCP packet to service " + svcName + " with type=LoadBalancer in namespace " + namespaceName + " from backend pod " + backendName)
+
+		_, err = curlInContainer(clientContainer, svcIP, endpointHTTPPort, "big.iso -o big.iso", 5)
+		framework.ExpectNoError(err, "failed to curl load balancer service")
+
+		ginkgo.By("change MTU on intermediary router to force icmp related packets")
+		cmd := []string{containerRuntime, "exec", routerContainer}
+		mtuCommand := strings.Split("ip link set mtu 1200 dev eth1", " ")
+
+		cmd = append(cmd, mtuCommand...)
+		_, err = runCommand(cmd...)
+		framework.ExpectNoError(err, "failed to change MTU on intermediary router")
+
+		ginkgo.By("by sending a TCP packet to service " + svcName + " with type=LoadBalancer in namespace " + namespaceName + " from backend pod " + backendName)
+
+		_, err = curlInContainer(clientContainer, svcIP, endpointHTTPPort, "big.iso -o big.iso", 5)
+		framework.ExpectNoError(err, "failed to curl load balancer service")
+	})
+
+	ginkgo.It("Should ensure load balancer service works with 0 node ports when ETP=local", func() {
+
+		err := framework.WaitForServiceEndpointsNum(f.ClientSet, namespaceName, svcName, 4, time.Second, time.Second*120)
+		framework.ExpectNoError(err, fmt.Sprintf("service: %s never had an enpoint, err: %v", svcName, err))
+
+		numberOfETPRules := pokeIPTableRules(backendNodeName, "OVN-KUBE-ETP")
+		framework.ExpectEqual(numberOfETPRules, 2)
+
+		numberOfETPRules = pokeIPTableRules(backendNodeName, "OVN-KUBE-EXTERNALIP")
+		framework.ExpectEqual(numberOfETPRules, 4)
+
+		ginkgo.By("by sending a TCP packet to service " + svcName + " with type=LoadBalancer in namespace " + namespaceName + " from backend pod " + backendName)
+
+		time.Sleep(time.Minute)
```

Comment:
This 1 min sleep is to ensure all setup is done correctly, its probably only in the order of seconds or milliseconds, but wanted to keep a minute to be safe. I can change it to 5 seconds or something in next iteration with review comments.